src/Microsoft.ML.Core/Prediction/ITrainer.cs

```diff
@@ -87,91 +81,60 @@ public interface ITrainer
         PredictionKind PredictionKind { get; }
```

Comment:
I *would* like to get rid of this, in future. It seems like this test on `ITrainer` for, say, binary classification could be just as easily done by doing something like `trainer is ITrainer<IBinaryClassifierPredictor>`. (This hypothetical `IBinaryClassifierPredictor` is a marker interface, akin to and possibly descending `IPredictorProducing<float>`). I think we need to do that anyway for once we nuke the concept of `Signature`.